common/reviews/api/terminal.api.md

```diff
@@ -4,9 +4,20 @@
 
 ```ts
 
+/// <reference types="node" />
+
 import type { Brand } from '@rushstack/node-core-library';
-import type { ITerminal } from '@rushstack/node-core-library';
 import { NewlineKind } from '@rushstack/node-core-library';
```

Comment:
(This topic should probably be addressed as a separate PR.)

Looking at the code, after this change `@rushstack/terminal`'s has only trivial dependencies on `node-core-library` (which is a pretty heavyweight package currently).  Eliminating this dependency would reduce the size of tools that don't import `node-core-library`, but perhaps more importantly it would avoid any version coupling between the two packages.

The specific imports are `Brand` (type only), `NewlineKind` (a non-const enum), and `Text`.  The `Text` API calls are pretty minimal:
- `Text.getNewline()` - originally the enum itself had the newline character, but the introduction of `NewlineKind.OsDefault` required this helper.
- A call to `Text.escapeRegExp()` for a pointlessly dynamic `RegExp` that also suppresses our lint rule:
    ```
    // eslint-disable-next-line @rushstack/security/no-unsafe-regexp
    this._newlineRegex = new RegExp(`${Text.escapeRegExp(terminalProvider.eolCharacter)}|\\n`, 'g');
    ```
- One call to `Text.convertToLf()`

However `node-core-library` is used internally by some tests and compile-time code. So an interesting question would be how to make it a devDependency, but protect against someone later accidentally importing the package.

Reply:
We could make a package that depends on both `@rushstack/node-core-library` and `@rushstack/terminal` that does the validation where there need to be matches between the two packages. Sure, it feels a little weird to have a package that contains only the unit tests, but it's not that odd and it helps with the package coupling.

Reply:
You mean an unpublished Rush project? That seems fine.